modules/nginx-ingress-controller/main.tf

```diff
@@ -0,0 +1,109 @@
+
+
+data "aws_eks_cluster" "this" {
+  name = var.eks_cluster_name
+}
+
+data "aws_eks_cluster_auth" "this" {
+  name = var.eks_cluster_name
+}
+
+provider "helm" {
+  kubernetes {
+    host                   = data.aws_eks_cluster.this.endpoint
+    cluster_ca_certificate = base64decode(data.aws_eks_cluster.this.certificate_authority.0.data)
+    token                  = data.aws_eks_cluster_auth.this.token
+    load_config_file       = false
+  }
+}
+
+provider "kubernetes" {
+  host                   = data.aws_eks_cluster.this.endpoint
+  cluster_ca_certificate = base64decode(data.aws_eks_cluster.this.certificate_authority.0.data)
+  token                  = data.aws_eks_cluster_auth.this.token
+  load_config_file       = false
+  version                = "~> 1.9"
+}
+
+
+
+resource "kubernetes_namespace" "nginx_ingress" {
```

Comment:
We can remove this resource entirely.

Instead, then set the following in the `helm_release` resource:

```terraform
namespace = var.kubernetes_namespace
create_namespace = true
```